### DISCUSSION TITLE
news url

### DISCUSSION THREAD

**Discussion by 3l3ktr0n** [e107inc/e107#4645] [Q&A Plugins] [answered]
my plugin is displaying news that are related to specific games...
currently news sef url is configured news/news-category/news-title
my plugin displays news/news-title for obvious reasons without the category, I guess.. it still works but I'd like to get the news-category also into the url for consistency

I've been trying here and there but fruitless..
now my question is.. how would I get news-category into the mix

```
function sc_news_title()
    {
        if($parm == 'no_url') { return $this->var["news_title"]; }
        if($parm == 'sef') { return $this->var["news_sef"]; }

        $urlparms = array(
            'news_id'     => $this->var["news_id"],
            'news_sef'    => $this->var['news_sef'],
        );

        return "<a href='".e107::getUrl()->create('news/view/item', $urlparms)."'>".$this->var["news_title"]."</a>";
    }
```

**Comment by 3l3ktr0n** [comment]
HAPPY NEW YEAR EVERYONE!

I used my gray cells and figured it out :)

**Reply by Moc** [reply]
Feel free to share the solution here, so the rest of the community can benefit as well! :)

**Comment by 3l3ktr0n** [comment, marked as answer]
Well it's nothing special actually. I was hoping the url just needs category_id and category_sef.. so I added them to the array. 

```
function sc_news_title()
    {
        if($parm == 'no_url') { return $this->var["news_title"]; }
        if($parm == 'sef') { return $this->var["news_sef"]; }

        $urlparms = array(
            'news_id'     => $this->var["news_id"],
            'news_sef'    => $this->var['news_sef'],
            'category_id'   => $this->var['category_id'],
            'category_sef'  => $this->var['category_sef'],
        );

        return "<a href='".e107::getUrl()->create('news/view/item', $urlparms)."'>".$this->var["news_title"]."</a>";
```

Later on made a query with JOIN to fetch them.

`$sql->gen("SELECT * FROM #news LEFT JOIN #news_category ON news_category = category_id WHERE news_id = {$NewsIds}");`

**Reply by Jimmi08** [reply]
I think that if you use 

`e107::getUrl()->create('news/view/item', $this->var)  `

the result would be the same
